stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

```diff
@@ -316,6 +316,41 @@ extension Unsafe${Mutable}RawBufferPointer: ${Mutable}Collection {
     }
     return 0
   }
+
+  public func _customIndexOfEquatableElement(_ element: UInt8) -> Index?? {
+    guard !isEmpty else { return Optional(nil) }
+
+    let ptr = baseAddress._unsafelyUnwrappedUnchecked
+    var i = 0
+
+    let stride = MemoryLayout<UInt>.stride
+    let address = Int(bitPattern: ptr)
+
+    while (address &+ i) % stride != 0 && i < count {
+      guard self[i] != element else { return Optional(i) }
+      i &+= 1
+    }
```

Comment:
We added API that makes this kind of arithmetic clearer:
```suggestion
    let aligned = ptr.alignedUp(for: UInt.self)
    while ptr.advanced(by: i) < aligned && i < endIndex {
      guard self[i] != element else { return Optional(i) }
      i &+= 1
    }
```

Reply:
fixed